cmd/kubeadm/app/util/config/masterconfig.go

```diff
@@ -56,9 +56,12 @@ func SetInitDynamicDefaults(cfg *kubeadmapi.InitConfiguration) error {
 	// This is the same logic as the API Server uses
 	ip, err := netutil.ChooseBindAddress(addressIP)
 	if err != nil {
-		return err
+		defaultBind := "127.0.0.1"
```

Comment:
The only problem I see with this is that ChooseBindAddress returns more than one error. We should really be returning a specific error type and only defaulting on the error type we expect otherwise we might ignore an actual error.

Reply:
thank makes sense.
but given it returns a string should we just string compare for "no default routes", like the unit tests are doing?

Reply:
I don't like comparing strings for errors, I'd prefer type checking as described https://blog.golang.org/error-handling-and-go

Reply:
ok, i will add the error type check.